2021-08-19 18:19:45,783 ERROR: Exception on /explore [GET] [in c:\code\personal\microblog\venv\lib\site-packages\flask\app.py:1454]
Traceback (most recent call last):
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Code\Personal\microblog\app\routes.py", line 165, in explore
    return render_template("index.html", title='Explore', posts=posts.items,
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "c:\code\personal\microblog\venv\lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "c:\code\personal\microblog\venv\lib\site-packages\jinja2\environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Code\Personal\microblog\app\templates\index.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "C:\Code\Personal\microblog\app\templates\base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block 'doc'
    {%- block html %}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block 'html'
    {% block body -%}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block 'body'
    {% block content -%}
  File "C:\Code\Personal\microblog\app\templates\base.html", line 87, in block 'content'
    {% block app_content %}{% endblock %}
  File "C:\Code\Personal\microblog\app\templates\index.html", line 11, in block 'app_content'
    {% include '_post.html' %}
  File "C:\Code\Personal\microblog\app\templates\_post.html", line 12, in top-level template code
    said {{ moment(post.timestamp).fromNow() }}:
  File "c:\code\personal\microblog\venv\lib\site-packages\jinja2\utils.py", line 84, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
jinja2.exceptions.UndefinedError: 'momoent' is undefined
2021-08-19 18:20:27,962 ERROR: Exception on /explore [GET] [in c:\code\personal\microblog\venv\lib\site-packages\flask\app.py:1454]
Traceback (most recent call last):
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Code\Personal\microblog\app\routes.py", line 165, in explore
    return render_template("index.html", title='Explore', posts=posts.items,
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
  File "c:\code\personal\microblog\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "c:\code\personal\microblog\venv\lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "c:\code\personal\microblog\venv\lib\site-packages\jinja2\environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Code\Personal\microblog\app\templates\index.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "C:\Code\Personal\microblog\app\templates\base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block 'doc'
    {%- block html %}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block 'html'
    {% block body -%}
  File "c:\code\personal\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block 'body'
    {% block content -%}
  File "C:\Code\Personal\microblog\app\templates\base.html", line 87, in block 'content'
    {% block app_content %}{% endblock %}
  File "C:\Code\Personal\microblog\app\templates\index.html", line 11, in block 'app_content'
    {% include '_post.html' %}
  File "C:\Code\Personal\microblog\app\templates\_post.html", line 12, in top-level template code
    said {{ moment(post.timestamp).fromNow() }}:
  File "c:\code\personal\microblog\venv\lib\site-packages\jinja2\utils.py", line 84, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
jinja2.exceptions.UndefinedError: 'momoent' is undefined
2021-08-19 18:21:53,254 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 16:56:22,230 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:55]
2021-09-08 20:35:02,828 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:40:15,320 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:41:11,074 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:44:13,140 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:44:29,088 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:45:36,820 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:45:45,857 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:46:27,671 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:47:00,283 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:48:25,853 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:50:25,257 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:50:41,562 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:52:25,688 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:53:10,781 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:54:00,492 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:54:50,733 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:55:08,722 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 20:55:21,232 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 21:15:05,792 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 21:17:53,844 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 21:18:01,564 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-08 21:20:06,421 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:22:50,431 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:24:57,026 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:26:08,418 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:27:06,338 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:31:59,137 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:33:08,117 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:35:08,279 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:37:00,124 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:37:20,745 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:37:29,609 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:39:11,112 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:42:45,169 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:44:17,683 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
2021-09-09 21:44:32,485 INFO: Microblog startup [in C:\Code\Personal\microblog\app\__init__.py:52]
